Mathlib/Algebra/Group/Translate.lean

```diff
@@ -0,0 +1,78 @@
+/-
+Copyright (c) 2024 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import Mathlib.Algebra.BigOperators.Pi
+import Mathlib.Data.Set.Pointwise.SMul
+
+/-!
+# Translation operator
+
+This file defines the translation of a function from a group by an element of that group.
+
+# Notation
+
+`τ a f` is notation for `translate a f`.
+-/
+
+open Function Set
+open scoped Pointwise
+
+variable {ι α β M G H R : Type*} [AddCommGroup G]
+
+/-- Translation of a function in a group by an element of that group.
+`τ a f` is defined as `x ↦ f (x - a)`.  -/
+def translate (a : G) (f : G → α) : G → α := fun x ↦ f (x - a)
+
+@[inherit_doc] notation "τ " => translate
```

Comment:
This probably shouldn't be global notation

Reply:
Do you have a good idea of what locale it should be in?

Reply:
I don't :/